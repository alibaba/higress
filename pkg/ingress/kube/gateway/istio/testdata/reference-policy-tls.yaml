apiVersion: gateway.networking.k8s.io/v1alpha2
kind: GatewayClass
metadata:
  name: higress
spec:
  controllerName: higress.io/gateway-controller
---
apiVersion: gateway.networking.k8s.io/v1alpha2
kind: Gateway
metadata:
  name: gateway
  namespace: higress-system
spec:
  addresses:
  - value: higress-gateway
    type: Hostname
  gatewayClassName: higress
  listeners:
  - name: cross
    hostname: "cert1.domain.example"
    port: 443
    protocol: HTTPS
    allowedRoutes:
      namespaces:
        from: Selector
        selector:
          matchLabels:
            kubernetes.io/metadata.name: "cert"
    tls:
      mode: Terminate
      certificateRefs:
      - name: cert
        namespace: cert
---
apiVersion: gateway.networking.k8s.io/v1alpha2
kind: ReferenceGrant
metadata:
  name: allow-cert
  namespace: cert
spec:
  from:
  - group: gateway.networking.k8s.io
    kind: Gateway
    namespace: higress-system
  to:
    - group: ""
      kind: Secret
---
apiVersion: gateway.networking.k8s.io/v1alpha2
kind: HTTPRoute
metadata:
  name: http
  namespace: cert
spec:
  parentRefs:
  - name: gateway
    namespace: higress-system
  hostnames: ["cert1.domain.example"]
  rules:
  - backendRefs:
    - name: httpbin
      port: 80
