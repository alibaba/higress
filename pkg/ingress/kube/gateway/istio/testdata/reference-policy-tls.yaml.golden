apiVersion: networking.istio.io/v1alpha3
kind: Gateway
metadata:
  annotations:
    internal.istio.io/gateway-service: higress-gateway.higress-system.svc.domain.suffix
    internal.istio.io/parents: Gateway/gateway/cross.higress-system
  creationTimestamp: null
  name: gateway-istio-autogenerated-k8s-gateway-cross
  namespace: higress-system
spec:
  servers:
  - hosts:
    - cert/cert1.domain.example
    port:
      name: default
      number: 443
      protocol: HTTPS
    tls:
      credentialName: kubernetes-gateway://cert/cert
      mode: SIMPLE
---
apiVersion: networking.istio.io/v1alpha3
kind: VirtualService
metadata:
  annotations:
    internal.istio.io/parents: HTTPRoute/http.cert
    internal.istio.io/route-semantics: gateway
  creationTimestamp: null
  name: http-0-istio-autogenerated-k8s-gateway
  namespace: cert
spec:
  gateways:
  - higress-system/gateway-istio-autogenerated-k8s-gateway-cross
  hosts:
  - cert1.domain.example
  http:
  - name: cert/http
    route:
    - destination:
        host: httpbin.cert.svc.domain.suffix
        port:
          number: 80
---
